design-documents/graph-ql/coverage/b2b/purchase-order.graphqls

```diff
@@ -0,0 +1,303 @@
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+type UpdatePurchaseOrderApprovalRuleOutput {
+    approval_rule: PurchaseOrderApprovalRule @doc(description: "Updated purchase order approval rule")
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+type DeletePurchaseOrderApprovalRuleOutput {
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+input CreatePurchaseOrderApprovalRuleInput {
+    approval_rule: PurchaseOrderApprovalRuleInput! @doc(description: "Purchase order approval rule data")
+}
+
+input UpdatePurchaseOrderApprovalRuleInput {
+    approval_rule_uid: ID! @doc(description: "Purchase order approval rule ID")
+    approval_rule: PurchaseOrderApprovalRuleInput! @doc(description: "Purchase order approval rule data")
+}
+
+input DeletePurchaseOrderApprovalRuleInput {
+    approval_rule_uid: ID! @doc(description: "Purchase order approval rule ID")
+}
+
+input PurchaseOrderApprovalRuleInput {
+    name: String! @doc(description: "Purchase order approval rule name")
+    description: String @doc(description: "Purchase order approval rule description")
+    applies_to: [ID]! @doc(description: "A list of B2B user roles to which this purchase order approval rule should be applied. In case when empty array is provided, the rule will be applied to all user roles in the system, including those created in the future")
+    type: PurchaseOrderApprovalRuleType! @doc(description: "Purchase order approval rule type")
+    condition: CreatePurchaseOrderApprovalRuleConditionInput! @doc(description: "Purchase order approval rule condition")
+    requires_approval_from: [ID!]! @doc(description: "A list of B2B user roles that can approve this purchase order approval rule")
+}
+
+input CreatePurchaseOrderApprovalRuleConditionInput {
+    operator: PurchaseOrderApprovalRuleConditionOperator! @doc(description: "Purchase order approval rule condition operator")
+    amount: CreatePurchaseOrderApprovalRuleConditionAmountInput @doc(description: "Purchase order approval rule condition ammount. Is mutually exclusive with condition quantity")
+    quantity: Int @doc(description: "Purchase order approval rule condition quantity. Is mutually exclusive with condition amount")
+}
+
+input CreatePurchaseOrderApprovalRuleConditionAmountInput {
+    value: Float! @doc(description: "Purchase order approval rule condition ammount value")
+    currency: CurrencyEnum! @doc(description: "Purchase order approval rule condition ammount currency")
+}
+
+input ApprovePurchaseOrderInput {
+    purchase_order_uid: ID!
```

Comment:
@DrewML do you suggest to call id field just uid in cases like this [here](https://github.com/magento/architecture/pull/415/files#diff-5073a4c5d52122c75aa507353b6f66b0R16)?

Reply:
@melnikovi Check out this section of the linked doc:

> **Terminology**
> - **Primary Identifier**: An ID owned by the current Object Type (ex: `Product.id`)
> - **Foreign Identifier**: An ID referencing another Object Type (ex: `PaymentMethodInput.code`)

In this case, `ApprovePurchaseOrderInput.purchase_order_uid` is a _foreign identifier_, so it should be covered by this:

> - Add a new field with a name of the format `{ForeignObject}_uid`, where `ForeignObject` is the name of the Object Type with the matching `uid` field